vita3k/gui/src/game_install_dialog.cpp

```diff
@@ -0,0 +1,124 @@
+// Vita3K emulator project
+// Copyright (C) 2019 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#include "interface.h"
+#include "private.h"
+
+#include <gui/functions.h>
+#include <gui/state.h>
+
+#include <host/functions.h>
+#include <host/sfo.h>
+#include <host/state.h>
+
+#include <io/vfs.h>
+
+#include <util/log.h>
+
+#include <nfd.h>
+
+namespace gui {
+
+nfdchar_t *game_path = nullptr;
+static bool draw_file_dialog = true;
+static bool game_install_confirm = false;
+
+void draw_game_install_dialog(GuiState &gui, HostState &host) {
+    nfdresult_t result = NFD_CANCEL;
+
+    if (draw_file_dialog) {
+        result = NFD_OpenDialog("vpk,zip", nullptr, &game_path);
+        draw_file_dialog = false;
+
+        if (result == NFD_OKAY) {
+            if (install_vpk(host, gui, static_cast<fs::path>(game_path)))
+                game_install_confirm = true;
+            else if (!gui.game_reinstall_confirm)
+                ImGui::OpenPopup("Game installation failed");
+        } else {
+            gui.file_menu.game_install_dialog = false;
+            draw_file_dialog = true;
+        }
+    }
+
+    static const auto BUTTON_SIZE = ImVec2(60.f, 0.f);
```

Comment:
static in function again is weird, scope is already hiding it

but i mean it does do something i think, it's just i feel like it would be better as a static global constant